core/scripts/chaincli/handler/scrape_node_config.go

```diff
@@ -0,0 +1,270 @@
+		}
+	}
+
+	if cnt != len(nodes) {
+		log.Infof("there are %d nodes provisioned , but .env is missing %d nodes", len(nodes), len(nodes)-cnt)
+	}
+}
+
+func (h *baseHandler) fetchNodeInfos(ctx context.Context, log logger.Logger) []NodeInfo {
+	client := http.DefaultClient
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, h.cfg.NodeConfigURL, nil)
+	if err != nil {
+		log.Fatalf("failed to build a GET request: %s", err)
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Fatalf("failed to make a GET request: %s", err)
+	}
+	defer resp.Body.Close()
+
+	var nodeInfos []NodeInfo
+	if err := json.NewDecoder(resp.Body).Decode(&nodeInfos); err != nil {
+		log.Fatalf("failed to read response: %s", err)
+	}
+
+	return nodeInfos
+}
+
+func (h *baseHandler) scrapeNodeInfo(wg *sync.WaitGroup, i int, cl cmd.HTTPClient, nodes map[string]*NodeInfo, log logger.Logger) {
+	defer wg.Done()
+
+	// get node addresses
+	resp, err := nodeRequest(cl, "/v2/keys/eth")
+	if err != nil {
+		log.Fatalf("failed to get ETH keys: %s", err)
+	}
+	var ethKeys cmd.EthKeyPresenters
+	if err = jsonapi.Unmarshal(resp, &ethKeys); err != nil {
+		log.Fatalf("failed to unmarshal response body: %s", err)
+	}
+	var nodeAddresses []string
+	for index := range ethKeys {
+		nodeAddresses = append(nodeAddresses, common.HexToAddress(ethKeys[index].Address).Hex())
+	}
+	ni := &NodeInfo{
+		NodeAddress: nodeAddresses,
+	}
+
+	// get node ocr2 config
+	ocr2Config, err := getNodeOCR2Config(cl)
+	if err != nil {
+		log.Fatalf("failed to get node OCR2 config: %s", err)
+	}
+	ni.Ocr2ID = []string{ocr2Config.ID}
+	ni.Ocr2OnchainPublicKey = []string{ocr2Config.OnchainPublicKey}
+	ni.Ocr2OffchainPublicKey = []string{ocr2Config.OffChainPublicKey}
+	ni.Ocr2ConfigPublicKey = []string{ocr2Config.ConfigPublicKey}
+
+	// get node p2p config
+	resp, err = nodeRequest(cl, "/v2/keys/p2p")
+	if err != nil {
+		log.Fatalf("failed to get p2p keys: %s", err)
+	}
+	var p2pKeys cmd.P2PKeyPresenters
+	if err = jsonapi.Unmarshal(resp, &p2pKeys); err != nil {
+		log.Fatalf("failed to unmarshal response body: %s", err)
+	}
+	ni.PeerId = []string{p2pKeys[0].PeerID}
+
+	// get node csa config
+	resp, err = nodeRequest(cl, "/v2/keys/csa")
+	if err != nil {
+		log.Fatalf("failed to get CSA keys: %s", err)
+	}
+	var csaKeys cmd.CSAKeyPresenters
+	if err = jsonapi.Unmarshal(resp, &csaKeys); err != nil {
+		log.Fatalf("failed to unmarshal response body: %s", err)
+	}
+	// this assumes the nodes are not multichain nodes and have only 1 node address assigned.
+	// for a multichain node, we can pass in a chain id and filter `ethKeys` array based on the chain id
+	// in terms of CSA keys, we need to wait for RTSP to support multichain nodes, which may involve creating one
+	// CSA key for each chain. but this is still pending so assume only 1 CSA key on a node for now.
+	csaKey := &CSAKeyInfo{
+		NodeAddress: nodeAddresses[0],
+		PublicKey:   csaKeys[0].PubKey,
```

Comment:
Should we add checks here to ensure there are one or more elements in each slice?

Reply:
done